x/ibc/24-host/validate.go

```diff
@@ -66,30 +81,61 @@ func PortIdentifierValidator(id string) error {
 func NewPathValidator(idValidator ValidateFn) ValidateFn {
 	return func(path string) error {
 		pathArr := strings.Split(path, "/")
+		if len(pathArr) > 0 && pathArr[0] == path {
+			return sdkerrors.Wrapf(ErrInvalidPath, "path %s doesn't contain any separator '/'", path)
+		}
+
+		allEmptyItems := true
 		for _, p := range pathArr {
-			// Each path element must either be valid identifier or alphanumeric
-			err := idValidator(p)
-			if err != nil && !sdk.IsAlphaNumeric(p) {
-				return sdkerrors.Wrapf(ErrInvalidPath, "path %s contains invalid identifier or non-alphanumeric path element: %s", path, p)
+			// NOTE: for some reason an empty string is added after Split
+			if p == "" {
+				continue
+			}
+
+			allEmptyItems = false
+
+			if err := idValidator(p); err != nil {
+				return err
+			}
+			// Each path element must either be a valid identifier or constant number
```

Comment:
Hmm this is not true by any path that ends in "/" unless the code elsewhere doesn't count the last empty string element as  a path element? ie `this/is/my/path/` elements: "this", "is", "my", "path", ""